src/git/tag.go

```diff
@@ -19,21 +19,39 @@ type TagsGetter interface {
 }
 
 type RepoTagsGetter struct {
-	workDir string
-	match   *regexp.Regexp
+	workDir      string
+	matchRegex   *regexp.Regexp
+	matchCommits map[string]bool
 }
 
 type TagOptionFunc func(s *RepoTagsGetter) error
 
-// TagsMatching returns an option that will make the getter to ignore tags that do not match regex.
-func TagsMatching(regex string) TagOptionFunc {
+// TagsMatchingRegex returns an option that will make the getter to ignore tags that do not match regex.
+func TagsMatchingRegex(regex string) TagOptionFunc {
```

Comment:
What is the motivation for this change? The name of the parameter here `regex string` should be already hinting what tags should match (a regex)

Reply:
If you are in the file `A` and you see a function call:
```
a.TagsMatching("abc")
```

Are you sure what is matching? Is is a string, a prefix or a regex? That was the reasoning behind 😅